Index: new/etc/vz.conf
===================================================================
--- new.orig/etc/vz.conf	2011-08-31 14:55:06.000000000 +0200
+++ new/etc/vz.conf	2011-08-31 14:58:29.000000000 +0200
@@ -36,8 +36,8 @@
 ## Defaults for containers
 VE_ROOT=/vz/root/$VEID
 VE_PRIVATE=/vz/private/$VEID
-CONFIGFILE="basic"
-DEF_OSTEMPLATE="centos-5"
+CONFIGFILE="pve.auto"
+DEF_OSTEMPLATE="debian-6.0-standard"
 
 ## Load vzwdog module
 VZWDOG="no"
