Index: new/etc/init.d/vzeventd-redhat.in
===================================================================
--- new.orig/etc/init.d/vzeventd-redhat.in	2012-02-10 06:33:24.000000000 +0100
+++ new/etc/init.d/vzeventd-redhat.in	2012-02-10 06:39:55.000000000 +0100
@@ -27,6 +27,11 @@
 prog=vzeventd
 lockfile=$VARLOCK/$prog
 
+if ! test -d /proc/vz ; then
+    # NOTE: Running kernel is not OpenVZ kernel.
+    exit 0
+fi
+
 load_module() {
 	modprobe vzevent reboot_event=1
 }
@@ -59,7 +64,7 @@
 	[ -f $lockfile ] && exit 0 # Already running
 
 	load_module
-	check || exit 1
+	check || exit 0
 
 	echo -n $"Starting $prog: "
 	vzdaemon_start $prog $OPTIONS
