Index: new/etc/vz.conf
===================================================================
--- new.orig/etc/vz.conf	2012-09-28 09:05:45.000000000 +0200
+++ new/etc/vz.conf	2012-09-28 09:38:27.000000000 +0200
@@ -33,8 +33,7 @@
 ## Defaults for containers
 VE_ROOT=/vz/root/$VEID
 VE_PRIVATE=/vz/private/$VEID
-CONFIGFILE="basic" # Use vswap-256m on RHEL6 kernel
-DEF_OSTEMPLATE="centos-5"
+
 ## Filesystem layout for new CTs: either simfs (default) or ploop
 #VE_LAYOUT=ploop
 
