Index: new/include/types.h
===================================================================
--- new.orig/include/types.h	2011-11-17 08:05:21.000000000 +0100
+++ new/include/types.h	2011-11-17 08:05:45.000000000 +0100
@@ -21,7 +21,7 @@
 #define VZ_DIR			PKGCONFDIR "/"
 #define GLOBAL_CFG		VZ_DIR "vz.conf"
 #define OSRELEASE_CFG		VZ_DIR "osrelease.conf"
-#define VPS_CONF_DIR		VZ_DIR "conf/"
+#define VPS_CONF_DIR		"/etc/pve/openvz/"
 #define LIB_SCRIPTS_DIR		PKGLIBDIR "/scripts/"
 #define DIST_DIR		VZ_DIR "dists"
 #define VENAME_DIR		VZ_DIR "names"
Index: new/paths.am
===================================================================
--- new.orig/paths.am	2011-11-17 08:05:21.000000000 +0100
+++ new/paths.am	2011-11-17 08:05:45.000000000 +0100
@@ -32,7 +32,7 @@
 ifupdir     = $(sysconfdir)/network/if-up.d
 
 pkgconfdir  = $(sysconfdir)/vz
-vpsconfdir  = $(pkgconfdir)/conf
+vpsconfdir  = $(sysconfdir)/pve/openvz
 distconfdir = $(pkgconfdir)/dists
 namesdir    = $(pkgconfdir)/names
 
Index: new/bin/vznetaddbr.in
===================================================================
--- new.orig/bin/vznetaddbr.in	2011-11-17 08:02:47.000000000 +0100
+++ new/bin/vznetaddbr.in	2011-11-17 08:05:45.000000000 +0100
@@ -2,7 +2,7 @@
 #
 # Add virtual network interfaces (veth's) in a container to a bridge on CT0
 
-CONFIGFILE=@PKGCONFDIR@/conf/$VEID.conf
+CONFIGFILE=@VPSCONFDIR@/$VEID.conf
 . $CONFIGFILE
 
 NETIFLIST=$(printf %s "$NETIF" |tr ';' '\n')
Index: new/bin/vzmigrate.in
===================================================================
--- new.orig/bin/vzmigrate.in	2011-11-17 08:02:47.000000000 +0100
+++ new/bin/vzmigrate.in	2011-11-17 08:05:45.000000000 +0100
@@ -31,7 +31,7 @@
 remove_area=1
 keep_dst=0
 debug=0
-confdir="@PKGCONFDIR@/conf"
+confdir="@VPSCONFDIR@"
 vzconf="@PKGCONFDIR@/vz.conf"
 tmpdir="/var/tmp"
 act_scripts=
Index: new/etc/init.d/vz-redhat.in
===================================================================
--- new.orig/etc/init.d/vz-redhat.in	2011-11-17 08:06:26.000000000 +0100
+++ new/etc/init.d/vz-redhat.in	2011-11-17 08:07:06.000000000 +0100
@@ -41,7 +41,7 @@
 . @VZLIBDIR@/initd-functions
 
 VZQUOTA=@SBINDIR@/vzquota
-CONFIG_DIR=@PKGCONFDIR@/conf
+CONFIG_DIR=@VPSCONFDIR@
 LOCKFILE=$VARLOCK/vz_lock
 SUBSYS_VZ=$VARLOCK/vz
 VESTAT=/proc/vz/vestat
