Index: new/etc/init.d/vz-redhat.in
===================================================================
--- new.orig/etc/init.d/vz-redhat.in	2012-09-28 09:39:27.000000000 +0200
+++ new/etc/init.d/vz-redhat.in	2012-09-28 09:42:11.000000000 +0200
@@ -25,8 +25,8 @@
 
 ### BEGIN INIT INFO
 # Provides: vz
-# Required-start: $network $remote_fs $syslog
-# Required-stop:  $network $remote_fs $syslog
+# Required-Start: +iscsi pve-cluster cman clvm $network $local_fs $remote_fs $syslog
+# Required-Stop:  +iscsi pve-cluster cman clvm $network $local_fs $remote_fs $syslog
 # Should-Start:  sshd vzeventd
 # Should-Stop:  sshd vzeventd
 # Default-Start: 2 3 4 5
@@ -40,6 +40,7 @@
 
 . @SCRIPTDIR@/initd-functions
 
+PVECTL=/usr/bin/pvectl
 VZQUOTA=@SBINDIR@/vzquota
 CONFIG_DIR=@VPSCONFDIR@
 LOCKFILE=$VARLOCK/vz_lock
@@ -290,10 +291,17 @@
 	local msg
 	local need_restart
 
+	sysctl -q -w net.ipv4.route.src_check=0
+
+	if [ -x ${PVECTL} ]; then
+	    # pve-manager should start the containers
+	    return;
+	fi
+
 	need_restart=""
 	velist=$(vzlist -aH -octid,onboot -s-bootorder |
 		awk '$2 == "yes" {print $1}')
-	sysctl -q -w net.ipv4.route.src_check=0
+
 	for veid in $velist; do
 		[ "${veid}" = "0" ] && continue
 		__echo "Starting CT ${veid}: "
