Index: new/bin/vznetaddbr.in
===================================================================
--- new.orig/bin/vznetaddbr.in	2011-10-05 09:03:07.000000000 +0200
+++ new/bin/vznetaddbr.in	2011-10-05 09:04:15.000000000 +0200
@@ -2,7 +2,7 @@
 #
 # Add virtual network interfaces (veth's) in a container to a bridge on CT0
 
-CONFIGFILE=@PKGCONFDIR@/conf/$VEID.conf
+CONFIGFILE=@VPSCONFDIR@/$VEID.conf
 . $CONFIGFILE
 
 NETIFLIST=$(printf %s "$NETIF" |tr ';' '\n')
