Index: new/bin/vznetaddbr.in
===================================================================
--- new.orig/bin/vznetaddbr.in	2013-10-04 09:09:45.000000000 +0200
+++ new/bin/vznetaddbr.in	2013-10-04 09:21:04.000000000 +0200
@@ -29,8 +29,14 @@
     [ -n "$bridge" ] ||
 	bridge=vmbr0
 
+    if [ ! -f "/sys/class/net/$bridge/mtu" ]; then
+	echo >&2 "bridge $bridge does not exists."
+	exit 1
+    fi
+
     echo "Adding interface $host_ifname to bridge $bridge on CT0 for CT$VEID"
-    ip link set dev "$host_ifname" up
+    bridge_mtu=$(cat "/sys/class/net/$bridge/mtu")
+    ip link set dev "$host_ifname" up mtu "$bridge_mtu"
     ip addr add 0.0.0.0/0 dev "$host_ifname"
     echo 1 >"/proc/sys/net/ipv4/conf/$host_ifname/proxy_arp"
     echo 1 >"/proc/sys/net/ipv4/conf/$host_ifname/forwarding"
