Index: new/src/lib/destroy.c
===================================================================
--- new.orig/src/lib/destroy.c	2012-09-28 09:05:45.000000000 +0200
+++ new/src/lib/destroy.c	2012-09-28 09:41:49.000000000 +0200
@@ -271,7 +271,7 @@
 	logger(0, 0, "Destroying container private area: %s", fs->private);
 	if ((ret = vps_destroy_dir(veid, fs->private)))
 		return ret;
-	move_config(veid, BACKUP);
+	move_config(veid, DESTR_ALL);
 	if (destroy_dumpfile(veid, cpt != NULL ? cpt->dumpdir : NULL) < 0)
 		logger(-1, errno, "Warning: failed to remove dump file");
 	if (rmdir(fs->root) < 0)
Index: new/include/util.h
===================================================================
--- new.orig/include/util.h	2012-09-28 09:05:45.000000000 +0200
+++ new/include/util.h	2012-09-28 09:41:49.000000000 +0200
@@ -24,6 +24,7 @@
 
 #define BACKUP		0
 #define DESTR		1
+#define DESTR_ALL	2
 
 #define PROCMEM		"/proc/meminfo"
 #define PROCTHR		"/proc/sys/kernel/threads-max"
Index: new/src/lib/util.c
===================================================================
--- new.orig/src/lib/util.c	2012-09-28 09:05:45.000000000 +0200
+++ new/src/lib/util.c	2012-09-28 09:41:49.000000000 +0200
@@ -613,7 +613,14 @@
 
 	snprintf(conf, sizeof(conf), VPS_CONF_DIR "%d.%s", veid, prefix);
 	snprintf(newconf, sizeof(newconf), "%s." DESTR_PREFIX, conf);
-	action == BACKUP ? rename(conf, newconf) : unlink(newconf);
+	if (action == BACKUP) {
+		rename(conf, newconf);
+	} else if (action == DESTR_ALL) {
+		unlink(conf);
+		unlink(newconf);
+	} else if (action == DESTR) {
+		unlink(newconf);
+	}
 }
 
 /* Renames or removes CT config and various CT scripts.
